### PR TITLE
Adding redis setup docs for running marqo locally

### PR DESCRIPTION
**Pull request by vicilliar**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
updated readme with redis setup

* **What is the current behavior?** (You can also link to an open issue here)
no current explanation for setting up redis baremetal

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no

* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -342,6 +342,35 @@ Marqo is a community project with the goal of making tensor search accessible to
 
 5. If you update dependencies, make sure to delete the .tox dir and rerun.
 
+## Redis set up
```
Comment (pandu-k) [inline README.md]: Developing locally doesn't actually require Redis, right? I've been able to run Marqo locally without redis running

Let's keep this README lightweight, and point possible contributors to the dev guide if they want a more detailed dev getting started.

File: src/marqo/README.md [hunk 2]
```diff
@@ -36,6 +36,35 @@ __Notes__:
 - This is for marqo-os (Marqo OpenSearch) running locally. You can alternatively set
 `OPENSEARCH_URL` to  a remote Marqo OpenSearch cluster 
 
+## Redis set up (Needed for Options A and E)
+Marqo uses redis to handle concurrency throttling. Redis is automatically set up when running Marqo in docker (Options B-D), but if you are running Marqo locally on your machine (Options A and E), you have to set redis up yourself.
```
Comment (pandu-k) [inline src/marqo/README.md]: We don't actually need Redis to be running for development right? It's more of an optional thing - if they want to play around with the throttling/have a closer experience to the final product

**Review by pandu-k — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (vicilliar) [reply README.md]: It's not, but I thought people might get confused as to why they get redis errors when they try to run Marqo locally without it.

**Comment by vicilliar** [comment]
unit tests passed: https://github.com/marqo-ai/marqo/actions/runs/4321253053

File: src/marqo/connections.py [hunk 3]
```diff
@@ -56,7 +56,7 @@ def init_from_app(self, host, port):
             logger.info(f"Took {((t1-t0)*1000):.3f}ms to connect to redis and load scripts.")
 
         except Exception as e:
-            logger.warn(f"There is a problem with your redis instance. Could not load throttling scripts onto redis. Reason: {e}")
+            logger.warn(f"There is a problem with your redis connection. Could not load throttling scripts onto redis. To suppress these warnings, disable throttling with export MARQO_ENABLE_THROTTLING='FALSE'. Read more under Redis setup section of the developer guide: https://github.com/marqo-ai/marqo/tree/mainline/src/marqo#developer-guide. Redis error reason: {e}")
```
Comment (pandu-k) [inline src/marqo/connections.py]: nit enclosing code within backticks helps readability: 
```
`export MARQO_ENABLE_THROTTLING='FALSE'` 
```

**Review by pandu-k — APPROVED** [review]

File: src/marqo/connections.py (hunk 3 above)
Comment (vicilliar) [reply src/marqo/connections.py]: Does this change the formatting in any way? Or is it just so they see the backticks?

**Comment by vicilliar** [comment]
trying unit tests: https://github.com/marqo-ai/marqo/actions/runs/4446323440/jobs/7806539562#step:8:5071

**Comment by vicilliar** [comment]
passed unit tests: https://github.com/marqo-ai/marqo/actions/runs/4446952956

ready for merge

**Review by pandu-k — APPROVED** [review]

**Pull request merged**